jax/_src/prng.py

```diff
@@ -1274,8 +1290,7 @@ def threefry_random_bits(key: typing.Array, bit_width, shape):
   if bit_width not in (8, 16, 32, 64):
     raise TypeError("requires 8-, 16-, 32- or 64-bit field width.")
 
-  if (config.jax_threefry_partitionable and
-      not any(core.is_special_dim_size(d) for d in shape)):
```

Comment:
This was a footgun: the behavior of the code would change if you have shape polymorphism